phpBB/viewonline.php

```diff
@@ -263,6 +263,55 @@
 		case 'viewtopic':
 			$forum_id = $row['session_forum_id'];
 
+			if (!$forum_id)
+			{
+				$array_match_index = 5;
+				// find the forum_id from the page URI
+				$matches = array();
+
+				$pattern = '#(viewtopic|posting)\.' . $phpEx . '((\?|&)\w+=\w+)*(\?|&)';
+				preg_match($pattern . 'f=(\d+)#', $row['session_page'], $matches, PREG_UNMATCHED_AS_NULL);
+				if ($matches[$array_match_index])
+				{
+					$forum_id = (int) $matches[$array_match_index];
+				}
+				else
+				{
+					preg_match($pattern . 't=(\d+)#', $row['session_page'], $matches, PREG_UNMATCHED_AS_NULL);
+					if ($matches[$array_match_index])
+					{
+						$topic_id = $matches[$array_match_index];
+
+						$sql = 'SELECT forum_id
+							FROM ' . TOPICS_TABLE . "
+							WHERE topic_id = $topic_id";
+						$db->sql_query($sql);
+						$forum_id = (int) $db->sql_fetchfield('forum_id');
```

Comment:
This will probably produce a bit of memory leak as the query result is not freed. Same below.

Reply:
Sorry about that.  A fix has been pushed.